Make WebCore::IDBSerializationContext testable

ddkilzer opened the pull request:
#### 76dc1d42c4219e4ae8beb7fb639f76382350fee8
<pre>
Make WebCore::IDBSerializationContext testable
<a href="https://bugs.webkit.org/show_bug.cgi?id=272032">https://bugs.webkit.org/show_bug.cgi?id=272032</a>
&lt;<a href="https://rdar.apple.com/125785355">rdar://125785355</a>&gt;

Reviewed by Sihui Liu.

Original patch by Jesse Hertz.

* Source/WebCore/Modules/indexeddb/IDBValue.h:
(WebCore::IDBValue::IDBValue):
* Source/WebCore/bindings/js/IDBBindingUtilities.h:
(WebCore::deserializeIDBValueToJSValue):
(WebCore::callOnIDBSerializationThreadAndWait):
- Export methods to make the code testable.

Canonical link: <a href="https://commits.webkit.org/277032@main">https://commits.webkit.org/277032@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30973e78d0d42b77dddc28aede0d39fd40d7e6ec

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46469 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25625 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49071 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49145 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42510 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48776 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29985 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23088 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/37918 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47047 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/22651 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40074 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19167 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/20033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41177 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4514 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/42799 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41553 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50981 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/21474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/17919 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/45167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/22766 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/44105 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/23167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6487 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22469 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (commented):
Can you make use of `callOnIDBSerializationThreadAndWait` for testing instead of making `IDBSerializationContext ` public?

Commit:
Make WebCore::IDBSerializationContext testable
https://bugs.webkit.org/show_bug.cgi?id=272032
<rdar://125785355>

Reviewed by Sihui Liu.

Original patch by Jesse Hertz.

* Source/WebCore/Modules/indexeddb/IDBValue.h:
(WebCore::IDBValue::IDBValue):
* Source/WebCore/bindings/js/IDBBindingUtilities.h:
(WebCore::deserializeIDBValueToJSValue):
(WebCore::callOnIDBSerializationThreadAndWait):
- Export methods to make the code testable.

Canonical link: https://commits.webkit.org/277032@main
